lib/pbench/test/unit/common/test_config.py

```diff
@@ -1,68 +1,252 @@
+        monkeypatch.setattr(ConfigParser, "get", MockedConfigParser.get)
+        config = PbenchConfig("pbench.cfg")
+        assert config.log_dir is None
+
+    def test_logger_dir(self, monkeypatch):
+        """verify logger_dir option in config file"""
+
+        class mock_MockedConfigParser(MockedConfigParser):
+            def __init__(self):
+                self._sections = {
+                    "logging": {"logger_type": "file", "log_dir": "/mock/log_dir"}
+                }
+
+        monkeypatch.setattr(configtools, "file_list", self.mock_file_list)
+        monkeypatch.setattr(ConfigParser, "__init__", mock_MockedConfigParser.__init__)
+        monkeypatch.setattr(ConfigParser, "get", MockedConfigParser.get)
+        config = PbenchConfig("pbench.cfg")
+        assert config.log_using_caller_directory is False
+        assert config.log_dir == "/mock/log_dir"
+
+    def test_no_logging_level_option(self, monkeypatch):
+        """test no logging_level option in config file"""
+
+        class mock_MockedConfigParser(MockedConfigParser):
+            def __init__(self):
+                self._sections = {
+                    "logging": {
+                        "logger_type": "file",
+                    }
+                }
+
+        monkeypatch.setattr(configtools, "file_list", self.mock_file_list)
+        monkeypatch.setattr(ConfigParser, "__init__", mock_MockedConfigParser.__init__)
+        monkeypatch.setattr(ConfigParser, "get", MockedConfigParser.get)
+        config = PbenchConfig("pbench.cfg")
+        assert config.default_logging_level == "INFO"
+
+    def test_logging_level(self, monkeypatch):
+        """verify logging_level option in config file"""
+
+        class mock_MockedConfigParser(MockedConfigParser):
+            def __init__(self):
+                self._sections = {
+                    "logging": {"logger_type": "file", "logging_level": "Devlog"}
+                }
+
+        monkeypatch.setattr(configtools, "file_list", self.mock_file_list)
+        monkeypatch.setattr(ConfigParser, "__init__", mock_MockedConfigParser.__init__)
+        monkeypatch.setattr(ConfigParser, "get", MockedConfigParser.get)
+        config = PbenchConfig("pbench.cfg")
+        assert config.default_logging_level == "Devlog"
+
+    def test_no_log_fmt_option(self, monkeypatch):
+        """test no log_format in config file"""
+
+        class mock_MockedConfigParser(MockedConfigParser):
+            def __init__(self):
+                self._sections = {
+                    "logging": {
+                        "logger_type": "file",
+                    }
+                }
+
+        monkeypatch.setattr(configtools, "file_list", self.mock_file_list)
+        monkeypatch.setattr(ConfigParser, "__init__", mock_MockedConfigParser.__init__)
+        monkeypatch.setattr(ConfigParser, "get", MockedConfigParser.get)
+        config = PbenchConfig("pbench.cfg")
+        assert config.log_fmt is None
```

Comment:
This test looks to me to be identical to `test_logger()` except for the assertion at line 233. I recommend that you move the assertion from here to that test and omit this test.

Reply:
yes